Funcky.Test/Extensions/ParseExtensionsTest.cs

```diff
@@ -14,53 +15,232 @@ public ParseExtensionsTest()
+        {
+            Assert.Equal(expected, input.ParseEnumOrNone<MyEnum>());
         }
+
+        private static TheoryData<Option<bool>, string> BooleanStrings()
+            => new TheoryData<Option<bool>, string>
+            {
+                { Option<bool>.None(), string.Empty },
+                { Option.Some(true), "true" },
+                { Option.Some(false), "false" },
+                { Option.Some(true), "TrUe" },
+                { Option.Some(false), "FalsE" },
+                { Option<bool>.None(), "0" },
+                { Option<bool>.None(), "1" },
+                { Option<bool>.None(), "T" },
+                { Option<bool>.None(), "F" },
+                { Option<bool>.None(), "falsch" },
+                { Option<bool>.None(), "bool" },
+                { Option<bool>.None(), "none" },
+            };
+
+        private static TheoryData<Option<byte>, string> ByteStrings()
+            => new TheoryData<Option<byte>, string>
+            {
+                { Option<byte>.None(), string.Empty },
+                { Option<byte>.None(), "no number" },
+                { Option.Some((byte)12), "+12" },
+                { Option.Some((byte)42), "42" },
+                { Option<byte>.None(), "One" },
+                { Option<byte>.None(), "+-1" },
+                { Option<byte>.None(), "MCMI" },
+                { Option<byte>.None(), "2000" },
+                { Option<byte>.None(), "1337E+02" },
+            };
+
+        private static TheoryData<Option<short>, string> ShortStrings()
+            => new TheoryData<Option<short>, string>
+            {
+                { Option<short>.None(), string.Empty },
+                { Option<short>.None(), "no number" },
+                { Option.Some((short)-12), "-12" },
+                { Option.Some((short)0), "0" },
+                { Option.Some((short)-9542), "-9542" },
+                { Option.Some((short)1337), "1337" },
+                { Option<short>.None(), "One" },
+                { Option<short>.None(), "+-1" },
+                { Option<short>.None(), "MCMI" },
+                { Option<short>.None(), "1337E+02" },
+            };
+
+        private static TheoryData<Option<int>, string> IntStrings()
+            => new TheoryData<Option<int>, string>
+            {
+                { Option<int>.None(), string.Empty },
+                { Option<int>.None(), "no number" },
+                { Option.Some(-12), "-12" },
+                { Option.Some(0), "0" },
+                { Option.Some(-953542), "-953542" },
+                { Option.Some(1337), "1337" },
+                { Option<int>.None(), "One" },
+                { Option<int>.None(), "+-1" },
+                { Option<int>.None(), "MCMI" },
+                { Option<int>.None(), "1337E+02" },
+            };
+
+        private static TheoryData<Option<long>, string> LongStrings()
+        => new TheoryData<Option<long>, string>
```

Comment:
Indentation. There are a couple of more cases in this file

Reply:
You could also use target typed new here I think

Reply:
Adressed in 1595a73c1dc9edd7f7ab8d5c08140918d613004b